### DISCUSSION TITLE
Showing a Toast for situation other than Error

### DISCUSSION THREAD

**Discussion by ABarLT** [huggingface/chat-ui#1451] [Q&A]
Does anyone know how to make a svelte toast pop-up in a scenario that does not involve throwing an error? I want the same type of informative toast error (like is already incorporated based on the error store) to pop-up on the screen to tell a user about some parsing logic that I am adding, without having to throw an error and stop the execution of the call to the model. Any help/insights would be much appreciated :)

**Comment by nsarrazin** [comment]
You could probably do it using a store, write the message you want to display in the store and have a component that reads the store that will reactively display the toast elements and resets the store

**Reply by nsarrazin** [reply]
If you feel like you have a feature idea that could be relevant for chat-ui, feel free to open an issue and someone with svelte experience could help out if it's a generally useful idea!

**Reply by ABarLT** [reply]
Hi! Thank you for getting back to me and for your help on this!! 
Here's a little more explanation of what I am trying to do. 

The feature that I am adding is the ability for users to upload documents of different types (PDF, PPTX, DOCX, MD) to use in their conversation with the model. I created an API that I call that does the parsing of the file and returns it as a string of text. Similar to how images are processed when uploaded before calling a model, I process the document before sending the text to the model. During the processing, I check to make sure that the returned string is not above a certain length, and if it is, the document is truncated to fit the max length constraints, but still carries on with sending what it can to the model (I don't want to halt execution by throwing an error, which is how the other Toast messages pop-up). A document being truncated is what I want to trigger the Toast message that lets the user know that their document is truncated, just so that they are aware their full document was not sent to the model. I have tried using a store, like you suggested. I have the writable store in the same file as the error store (errors.ts), and follow the same logic for it as the error store in the src/routes/+layout.svelte file. I set the store for the truncation message in the logic of checking the length of the parsed string. However, this does not trigger the Toast from the layout file, which I believe is because the setting of the store is not in the scope of the layout file. Given this structure of my code, do you think that showing the Toast in that context would work in some other way?

**Reply by ABarLT** [reply]
I decided to opt for showing a Toast in the same manner that is done throughout the app already (throwing an error), as a response generated for a truncated document would not be sufficient for a user's needs anyways.
